packages/commonwealth/server/migrations/20220822175427-fix-entity-duplicates.js

```diff
@@ -0,0 +1,418 @@
+'use strict';
+
+module.exports = {
+    // WARNING: irreversible migration
+    // WARNING: Migration assumes that `ts-node /server/scripts/getEntitiesWithEvents.ts` was already executed successfully
+    up: async (queryInterface, Sequelize) => {
+        const [temp, metadata] = (await queryInterface.sequelize.query(`
+            SELECT EXISTS(
+                           SELECT
+                           FROM pg_tables
+                           WHERE schemaname = 'public'
+                             AND tablename = 'entities_creation_events'
+                       );
+        `, {raw: true}));
+        if (!temp[0].exists) throw new Error("The 'entities_creation_events' table is not defined. Cannot run migration! " +
+            "Please run `cd server/scripts && ts-node getEntitiesWithEvents.ts && cd ../..` first.");
+
+        return queryInterface.sequelize.transaction(async (t) => {
+            /**
+             * Entities with no chain-events
+             */
+
+            await queryInterface.sequelize.query(`
+                SELECT *
+                INTO delete_no_event_entities
+                FROM "ChainEntities"
+                WHERE NOT EXISTS(
+                        SELECT FROM "ChainEvents" WHERE entity_id = "ChainEntities".id
+                    );
+            `, {transaction: t});
+
+
+            await queryInterface.sequelize.query(`
+                DELETE
+                FROM "ChainEntities"
+                WHERE id IN (SELECT id FROM delete_no_event_entities);
+            `, {transaction: t});
+
+            await queryInterface.sequelize.query(`
```

Comment:
How many comments do we end up deleting -- anything significant?

Reply:
Very minimal - in this case only 4 tho the query is not hardcoded in order to account for any comments that may be created before this migration is merged and deployed.